packages/rules.lua

```diff
@@ -43,25 +43,72 @@ SILE.registerCommand("underline", function (_, content)
   local underlineThickness = font.post.underlineThickness / upem * fontoptions.size
 
   local hbox = SILE.call("hbox", {}, content)
-  local gl = SILE.length() - hbox.width
-  SILE.call("lower", {height = SU.cast("measurement", underlinePosition)}, function()
-    SILE.call("hrule", {width = gl.length, height = underlineThickness})
-  end)
-  SILE.typesetter:pushGlue({width = hbox.width})
+  table.remove(SILE.typesetter.state.nodes) -- steal it back...
+
+  -- Re-wrap the hbox in another hbox responsible for boxing it at output
+  -- time, when we will know the line contribution and can compute the scaled width
+  -- of the box, taking into account possible stretching and shrinking.
+  SILE.typesetter:pushHbox({
+    inner = hbox,
+    width = hbox.width,
+    height = hbox.height,
+    depth = hbox.depth,
+    outputYourself = function(self, typesetter, line)
+      local oldX = typesetter.frame.state.cursorX
+      local Y = typesetter.frame.state.cursorY
+
+      -- Build the original hbox.
+      -- Cursor will be moved by the actual definitive size.
+      self.inner:outputYourself(SILE.typesetter, line)
+      local newX = typesetter.frame.state.cursorX
+
+      -- Output a line.
+      -- NOTE: According to the OpenType specs, underlinePosition is "the suggested distance of
+      -- the top of the underline from the baseline" so it seems implied that the thickness
+      -- should expand downwards
+      SILE.outputter:drawRule(oldX, Y + underlinePosition, newX - oldX, underlineThickness)
+    end
+  })
 end, "Underlines some content")
 
 SILE.registerCommand("boxaround", function (_, content)
+  -- This command was not documented and lacks feature.
+  -- Plan replacement with a better suited package.
+  SU.deprecated("\\boxaround (undocumented)", "\\framebox (package)", "0.12.0", "0.13.0")
```

Comment:
This bit needs splitting out into a different commit for the sake of the release notes. Deprecations stuffed inside `fix:` commits won't get the airtime they require.

Reply:
P.S. I don't mind doing the split, I have `git revise -c` + interactive diff edits and am not afraid to use them. While we're in here I'd fix the underline and boxaround commands in two commits, again for the automatic release note benefits.

Reply:
You were quicker than I :)
Great, thanks !

Reply:
Oops the commit messages say "borderbox" for actually "boxaround"